test/test.ads.js

```diff
@@ -1169,3 +1169,39 @@ QUnit.test('adserror ends linear ad mode ', function(assert) {
   this.player.trigger('adserror');
   assert.strictEqual(this.player.ads._inLinearAdMode, false, 'after adserror');
 });
+
+if (videojs.browser.IS_IOS) {
+  QUnit.test('Check the trackChangeDuringAdHandler takes effect on iOS', function(assert) {
+    const tracks = this.player.textTracks();
+
+    // Since addTextTrack is async, wait for the addtrack event
+    tracks.on('addtrack', function() {
+
+      // Confirm the track is added, set the mode to showing
+      assert.equal(tracks.length, 1);
+      tracks[0].mode = 'showing';
+      assert.equal(tracks[0].mode, 'showing',  'Initial state is showing');
+
+      // Start the ad, confirm the track is disabled
+      this.player.ads.startLinearAdMode();
+      assert.equal(tracks[0].mode, 'disabled', 'Snapshot sets tracks to disabled');
+
+      // Force the mode to showing
+      tracks[0].mode = 'showing';
+
+    }.bind(this));
+
+    // The mode should go back to disabled when the change happens
```

Comment:
Since this issue is subtle, it could be helpful to be more specific about what the change is.